docs/src/modules/components/DemoSandboxed.js

```diff
@@ -80,26 +117,98 @@ DemoFrame.propTypes = {
 
 const StyledFrame = withStyles(styles, { withTheme: true })(DemoFrame);
 
+const sandboxedStyles = theme => ({
+  errorIcon: {
+    margin: theme.spacing(0, 1),
+    marginBottom: -3,
+  },
+});
+
 /**
  * Isolates the demo component as best as possible. Additional props are spread
  * to an `iframe` if `iframe={true}`.
  */
-function DemoSandboxed(props) {
-  const { component: Component, iframe, name, ...other } = props;
-  const Sandbox = iframe ? StyledFrame : React.Fragment;
-  const sandboxProps = iframe ? { title: `${name} demo`, ...other } : {};
-
-  return (
-    <Sandbox {...sandboxProps}>
-      <Component />
-    </Sandbox>
-  );
+class DemoSandboxed extends React.PureComponent {
+  state = {
+    hasError: false,
+  };
+
+  static getDerivedStateFromError() {
+    return { hasError: true };
+  }
+
+  componentDidCatch(error) {
+    // eslint-disable-next-line no-console
+    console.error(error);
+  }
+
+  render() {
+    const { classes, component: Component, iframe, name, ...other } = this.props;
+    const { hasError } = this.state;
+
+    if (hasError) {
+      const title = `[docs] Demo ${name} crashes`;
+      const searchQuery = encodeURIComponent(`is:issue ${title}`);
+      const issueLink = newGithubIssueUrl({
+        user: 'mui-org',
+        repo: 'material-ui',
+        title,
+        body: `
+<!-- Please make sure you have fullfilled the following items before submitting -->
+<!-- Checked checkbox should look like this: [x] -->
+- [ ] I have [searched for similar issues](https://github.com/mui-org/material-ui/issues?q=${searchQuery}) of this repository and believe that this is not a duplicate.
+
+## Steps to Reproduce
+1. Visit ${window.location.href}
+2. ??
+3. demo *${name}* crashes
+
+## Your Environment
+| Tech         | Version |
+|--------------|---------|
+| Material-UI  | v${process.env.LIB_VERSION}  |
+| Browser      | ${
+          typeof window !== 'undefined' && window.navigator
+            ? window.navigator.userAgent
+            : '*Unknown*'
+        } |
+`,
+      });
+
+      /* eslint-disable material-ui/no-hardcoded-labels */
+      return (
+        <div>
+          <Typography color="error" component="p" variant="h5">
+            <ErrorIcon className={classes.errorIcon} />
+            This demo had a runtime error
+          </Typography>
+          <Typography>
+            We would appreciate it if you report this error directly to our issue tracker. By
```

Comment:
```suggestion
            We would appreciate it if you <Link href={issueLink}>report this error<Link> directly in our issue tracker.
```

Reply:
Perfect. I was struggling to find a good wording to convert the action into a link.